epson_print_conf.py

```diff
@@ -25,6 +33,33 @@ class EpsonPrinter:
     """SNMP Epson Printer Configuration."""
 
     PRINTER_CONFIG = {  # Known Epson models
+        "L386": {
+            'borderless_waste': {'divider': 24.2, 'oids': [26, 27, 34]},
+            'main_waste': {'divider': 62.07, 'oids': [24, 25, 30]},
+            'raw_waste_reset': {24: 0,
+                                25: 0,
+                                26: 0,
+                                27: 0,
+                                28: 0,
+                                29: 0,
+                                30: 0,
+                                34: 0,
+                                46: 94,
+                                47: 94,
+                                49: 0},
+            'read_key': [16, 8],
+            'serial_number': range(192, 202),
+            'stats': {
+                'First TI received time': [173, 172],
+                'Manual cleaning counter': [147],
+                'Power cleaning counter': [148],
+                'Timer cleaning counter': [149],
+                'Total print page counter': [167, 166, 165, 164],
+                'Total print pass counter': [171, 170, 169, 168],
+                'Total scan counter': [471, 470, 469, 468]
+            },
+            'write_key': b'Sinabung'
+        },
```

Comment:
This should be more complete:

```python
        "L386": {
            "read_key": [16, 8],
            "write_key": b"Sinabung",
            "main_waste": {"oids": [24, 25, 30], "divider": 62.07},
            "borderless_waste": {"oids": [26, 27, 34], "divider": 24.2},
            "raw_waste_reset": {
                24: 0, 25: 0, 30: 0,  # Data of 1st counter
                28: 0, 29: 0,  # another store of 1st counter
                46: 94,  # Maintenance required level of 1st counter
                26: 0, 27: 0, 34: 0,  # Data of 2nd counter
                47: 94,  # Maintenance required level of 2st counter
                49: 0  # ?
            },
            "stats": {
                "Manual cleaning counter": [147],
                "Timer cleaning counter": [149],
                "Power cleaning counter": [148],
                "Total print pass counter": [171, 170, 169, 168],
                "Total print page counter": [167, 166, 165, 164],
                "Total scan counter": [471, 470, 469, 468],
                "First TI received time": [173, 172],
                "Maintenance required level of 1st waste ink counter": [46],
                "Maintenance required level of 2nd waste ink counter": [47],
            },
            "serial_number": range(192, 202),
            "last_printer_fatal_errors": [
                453, 452, 455, 454, 457, 456, 459, 458, 461, 460, 467,
                499, 498, 501, 500, 503, 502, 505, 504, 507, 506
            ],
        },
```

The printer looks similar to XP-315, ET-2600, ET-2650, L395.

You might also test "Power off timer", "printer_head_id" and "ink_replacement_counters" from XP-315 with this code:

```python
        "L386": {
            "alias": ["XP-315"],
            "read_key": [16, 8],
            "write_key": b"Sinabung",
            "main_waste": {"oids": [24, 25, 30], "divider": 62.07},
            "borderless_waste": {"oids": [26, 27, 34], "divider": 24.2},
            "last_printer_fatal_errors": [
                453, 452, 455, 454, 457, 456, 459, 458, 461, 460, 467,
                499, 498, 501, 500, 503, 502, 505, 504, 507, 506
            ],
        },
```

Reply:
> Unless I'm misunderstanding something - L386 is an inktank printer so it doesn't have ink replacement counters

OK, let's add the appropriate features (copying them from XP-315) instead of pointing to "XP-315".

> Printer_head_id comes back as "038E8E71 - 0D", how do I know if this is correct?

You should find it on a small label on the head.

> Power off timer seems to work, but in the code there seems to be self.parm["stats"]["poweroff_timer"][0], "poweroff_timer" should be "Power off timer". I will add this in my PR.

I think that, the way options are managed in the code, features with spaces inside might not work. That was the reason of labels like "poweroff_timer" if I remember well.

Reply:
I didn't want to dissasemble my printer to find the label but in wicreset the head id is reported as "38E8E71P03", but i can see in the logs that it read the same exact addresses so I assume that is the issue with how we decode the bytes. I too don't understand how "0x0D" gets converted to "P03".

Reply:
This should go that way by now, I think the printer is appropriately configured now